### PR TITLE
feat: Add seller info view to product details 🍿

### PR DESCRIPTION
**Pull request by oradwanomar**
Screen Shot 2022-07-21 at 12 12 44 PM

### CODE REVIEW

File: UIDokan/Sources/UIDokan/Extensions/UILabel+Extension.swift [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+//
+
+import UIKit
+
+public extension UILabel {
+    
+    func addTrailing(image: UIImage, text:String) {
+        let attachment = NSTextAttachment()
+        attachment.image = image
+        attachment.bounds = CGRect(x: 0, y: -5, width: (attachment.image?.size.width)!, height: (attachment.image?.size.height)!)
```
Comment (ahmdmhasn) [inline UIDokan/Sources/UIDokan/Extensions/UILabel+Extension.swift]: Force unwrap is prohibited! You owe a Mega Ice Cream 🍦😁

I like the idea behind this TBH ❤️
- You may add size as an input to the method. 
- How y is -5.. It's not a frame and should be zero

Comment (ahmdmhasn) [reply UIDokan/Sources/UIDokan/Extensions/UILabel+Extension.swift]: Please document this method this it's mandatory to tell the developer that you're updating the attributed string.

File: Dokan/Dokan/Scenes/ProductDetails/Views/InfoSellerView/InfoSellerView.swift [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        super.init(coder: coder)
+        commonInit()
+    }
+    
+    private func commonInit(){
+        Bundle.main.loadNibNamed("InfoSellerView", owner: self, options: nil)
+        addSubview(contentView)
+        contentView.frame = self.bounds
+        contentView.autoresizingMask = [.flexibleHeight, .flexibleWidth]
+        sellerType.addTrailing(image: UIImage(named: "verified")!, text: sellerType.text! + " ")
```
Comment (ahmdmhasn) [inline Dokan/Dokan/Scenes/ProductDetails/Views/InfoSellerView/InfoSellerView.swift]: Never force unwrap.. The app will crash then users complaint :(

File: Dokan/Dokan/Scenes/ProductDetails/Views/InfoSellerView/InfoSellerView.swift [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    
+    @IBOutlet var contentView: UIView!
+    
+    @IBOutlet weak var sellerImageProfile: UIImageView!
+    
+    @IBOutlet weak var sellerName: UILabel!
+    
+    @IBOutlet weak var sellerType: UILabel!
+    
+    @IBOutlet weak var sellerNavigationButton: UIButton!
```
Comment (ahmdmhasn) [inline Dokan/Dokan/Scenes/ProductDetails/Views/InfoSellerView/InfoSellerView.swift]:
```suggestion
    
    // MARK: - Outlets
    
    @IBOutlet private weak var contentView: UIView!
    @IBOutlet private weak var sellerImageProfile: UIImageView!
    @IBOutlet private weak var sellerName: UILabel!
    @IBOutlet private weak var sellerType: UILabel!
    @IBOutlet private weak var sellerNavigationButton: UIButton!
```

File: Dokan/Dokan/Scenes/ProductDetails/ProductDetailsViewController.swift [hunk 4]
```diff
@@ -43,9 +44,14 @@ extension ProductDetailsViewController {
 //
 extension ProductDetailsViewController {
     
+    func configureInfoSellerNavButton(){
+        InfoSellerView.sellerNavigationButton.addTarget(self, action: #selector(infoSellerWasTapped), for: .touchUpInside)
+    }
+    
 }
 
 // MARK: - Private Handlers
```
Comment (ahmdmhasn) [inline Dokan/Dokan/Scenes/ProductDetails/ProductDetailsViewController.swift]:
```suggestion
// MARK: - Actions
```

File: Dokan/Dokan/Scenes/ProductDetails/ProductDetailsViewController.swift [hunk 5]
```diff
@@ -11,6 +11,7 @@ class ProductDetailsViewController: UIViewController {
     
     // MARK: Outlets
 
+    @IBOutlet weak var InfoSellerView: InfoSellerView!
```
Comment (ahmdmhasn) [inline Dokan/Dokan/Scenes/ProductDetails/ProductDetailsViewController.swift]:
```suggestion
    @IBOutlet private weak var InfoSellerView: InfoSellerView!
    
```

File: Dokan/Dokan/Scenes/ProductDetails/ProductDetailsViewController.swift [hunk 6]
```diff
@@ -43,9 +44,14 @@ extension ProductDetailsViewController {
 //
 extension ProductDetailsViewController {
     
+    func configureInfoSellerNavButton(){
```
Comment (ahmdmhasn) [inline Dokan/Dokan/Scenes/ProductDetails/ProductDetailsViewController.swift]: Not called?

File: Dokan/Dokan/Assets.xcassets/dumy.imageset/Contents.json [hunk 7]
```diff
@@ -0,0 +1,21 @@
+{
```
Comment (ahmdmhasn) [inline Dokan/Dokan/Assets.xcassets/dumy.imageset/Contents.json]: Pull request is a production-ready code.. We don't add dummy images 👻

File: Dokan/Dokan/Scenes/ProductDetails/Views/InfoSellerView/InfoSellerView.xib [hunk 8]
```diff
@@ -0,0 +1,121 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (ahmdmhasn) [inline Dokan/Dokan/Scenes/ProductDetails/Views/InfoSellerView/InfoSellerView.xib:1]: A lot of constraints with fixed constant values which will not fit for different screen sizes. We need to consider enhancing the view before merging it.

**Review by ahmdmhasn — COMMENTED** [review]
👻

**Comment by ahmdmhasn** [comment]
Rename title to `feat: Add seller info view to product details 🍿`

File: Dokan/Dokan/Scenes/ProductDetails/Views/InfoSellerView/InfoSellerView.swift [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+        commonInit()
+        addTapGestureRecognizer()
+    }
+    
+    private func commonInit(){
+        Bundle.main.loadNibNamed("InfoSellerView", owner: self, options: nil)
+        addSubview(contentView)
+        contentView.frame = self.bounds
+        contentView.autoresizingMask = [.flexibleHeight, .flexibleWidth]
+        sellerType.addTrailing(image: UIImage(named: "verified"), text: sellerType.text! + " ")
```
Comment (ahmdmhasn) [inline Dokan/Dokan/Scenes/ProductDetails/Views/InfoSellerView/InfoSellerView.swift]: - Please stop force unwrap. 
- I think this should be moved to configure method?

```suggestion
        sellerType.addTrailing(image: UIImage(named: "verified"), text: sellerType.text ?? "" + " ")
```

File: Dokan/Dokan/Scenes/ProductDetails/Views/InfoSellerView/InfoSellerView.swift [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+
+class InfoSellerView: UIView {
+    
+    @IBOutlet var contentView: UIView!
+    
+    @IBOutlet weak var sellerImageProfile: UIImageView!
+    
+    @IBOutlet weak var sellerName: UILabel!
+    
+    @IBOutlet weak var sellerType: UILabel!
```
Comment (ahmdmhasn) [inline Dokan/Dokan/Scenes/ProductDetails/Views/InfoSellerView/InfoSellerView.swift]:
```suggestion
    // MARK: Outlets
    
    @IBOutlet private weak var contentView: UIView!
    @IBOutlet private weak var sellerImageView: UIImageView!
    @IBOutlet private weak var sellerNameLabel: UILabel!
    @IBOutlet private weak var sellerTypeLabel: UILabel!
```

File: UIDokan/Sources/UIDokan/Extensions/UILabel+Extension.swift [hunk 11]
```diff
@@ -0,0 +1,37 @@
+//
+//  File.swift
+//  
+//
+//  Created by Omar Ahmed on 21/07/2022.
+//
+
+import UIKit
+
+public extension UILabel {
```
Comment (ahmdmhasn) [inline UIDokan/Sources/UIDokan/Extensions/UILabel+Extension.swift:10]: Please never force unwrap. You owe one ice-cream. 🍦

**Review by ahmdmhasn — COMMENTED** [review]
Neat and clean. Just come comments 🔗

**Review by ziadali22 — APPROVED** [review]

**Review by oradwanomar — COMMENTED** [review]
Review & Approve ya raniaaaaa @RaniaZeid

File: Dokan/Dokan/Scenes/ProductDetails/Views/InfoSellerView/InfoSellerView.swift [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+    
+    private func commonInit(){
+        Bundle.main.loadNibNamed("InfoSellerView", owner: self, options: nil)
+        addSubview(contentView)
+        contentView.frame = self.bounds
+        contentView.autoresizingMask = [.flexibleHeight, .flexibleWidth]
+        sellerType.addTrailing(image: UIImage(named: "verified"), text: sellerType.text ?? "" + " ")
+    }
+    
+    private func addTapGestureRecognizer(){
```
Comment (ahmdyasser) [inline Dokan/Dokan/Scenes/ProductDetails/Views/InfoSellerView/InfoSellerView.swift]:
```suggestion
    private func addTapGestureRecognizer() {
```

**Review by RaniaZeid — APPROVED** [review]

**Pull request merged**